src/extension/iSamples_csvExport.js

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState, useRef } from "react";
+import { solrQuery } from "solr-faceted-search-react";
+import { store } from "redux/store";
+import { CSVLink } from "react-csv";
+import cx from "classnames";
+
+const MAX_ROWS = 100000;
+
+const CsvExport = (props) => {
+  const { bootstrapCss } = props;
+  const [isLoading, setIsLoading] = useState(false);
+  const [collapse, setCollapse] = useState(false);
+  const [downloadData, setDownloadData] = useState("");
+  const [downloadStatus, setDownloadStatus] = useState("");
+  const [formInfo, setFormInfo] = useState({
+    start: 0,
+    rows: MAX_ROWS
+  })
+  const csvRef = useRef();
+
+  // Generate filename based on the query parameters
+  const filename = `${store.getState()['query']['searchFields']
+    .filter(field => field.value)
+    ?.map(field => field.value)
+    .flat(2)
+    .concat([formInfo.start, store.getState()['results']['numFound'] > formInfo.rows ? formInfo.rows : store.getState()['results']['numFound']])
+    .map(e => {
+      if (typeof e === "object") {
+        return "bounding_box"
+      }
+
+      return e;
+    })
+    .join("_")}.csv`;
```

Comment:
Generate filename based on facet query parameters. For the bounding box, the value is not readable. I just use `bounding_box` to replace the value.

Reply:
There may be some values that result in invalid file names, so should filter out path reserved characters, like `*`, `.`, `/`, etc